pkg/webui/account/containers/profile-settings-form/index.js

```diff
@@ -291,16 +299,15 @@ const ProfileEditForm = () => {
       />
       <SubmitBar>
         <Form.Submit component={SubmitButton} message={sharedMessages.saveChanges} />
-        <ModalButton
-          type="button"
-          icon="delete"
-          danger
-          naked
+        <DeleteModalButton
           message={m.deleteAccount}
-          modalData={{
-            message: m.modalWarning,
-          }}
+          entityId={userId}
+          entityName={user.name}
           onApprove={handleDelete}
+          shouldConfirm
+          mayPurge={mayPurge}
+          defaultMessage={m.deleteWarning}
+          purgeMessage={m.purgeWarning}
```

Comment:
The message needs to be a bit different in case of deleting one's own account.

```
Are you sure you want to delete your account?

This will <strong>PERMANENTLY DELETE YOUR ACCOUNT</strong> and make associated entities (e.g. gateways, applications and end devices) owned by you that do not have other collaborators <strong>UNACCESSIBLE</strong>. It will also <strong>NOT BE POSSIBLE TO REUSE THE ACCOUNT ID AND REGISTER ENTITIES WITH THE SAME ID OR EUI's</strong>. Make sure you assign new collaborators to such entities if you plan to continue using them.

Please enter your user ID to confirm the account deletion.
```